include/windows/osd.h

```diff
@@ -1000,11 +1001,34 @@ OFI_DEF_COMPLEX(long_double)
 /* atomics primitives */
 #ifdef HAVE_BUILTIN_ATOMICS
 #define InterlockedAdd32 InterlockedAdd
+#define InterlockedCompareExchange32 InterlockedCompareExchange
 typedef LONG ofi_atomic_int_32_t;
 typedef LONGLONG ofi_atomic_int_64_t;
 
+#define OFI_DEf_ATOMIC_WAPPERS(radix)							\
```

Comment:
Please rename to avoid the odd lowercase 'f'.  Though I wonder if we really need this macro, versus just declaring things directly.  Maybe try this:

```
#define ofi_atomic_cas_bool(radix, ptr, expected, desired) \
    return (InterlockedCompareExchange##radix(blah blah blah) == expected)
```

Reply:
Oh sorry, dunno how the lowercase 'f' came in, best and lame excuse I have, is that it is not a good idea to code on Saturday morning with my 3 year old my lap.
From ofi_atom.h and osd.h I thought inlined functions macros are preferred over plain macros...